docs/advanced/Channels.md

```diff
@@ -161,25 +161,38 @@ export function* saga() {
 }
 ```
 
-Here is another example of how you can use event channels to wait for events. Suppose you are waiting for an event that will trigger some event handler `onEvent`, which has yet to be defined. You'll want to define your `onEvent` function within your subscriber function:
+Here is another example of how you can use event channels to pass WebSocket events into your saga (using socket.io library). Suppose you are waiting for a server message 'ping' which you would like to consume with saga. You'll want to setup your subscription within your subscriber function:
 
 ```javascript
 import { take, put, call } from 'redux-saga/effects'
-import { eventChannel, END } from 'redux-saga'
+import { eventChannel, delay } from 'redux-saga'
+import socket from './socketConnection' // already established and exported connection
 
-export function* saga() {
+function* pong() {
+  yield call(delay, 5000)
+  socket.emit('pong')
+}
+
+export function* watchOnPings() {
+  const socketChannel = eventChannel(emit => {
```

Comment:
It's not possible to test this form. Can we extract this into a separate a function like `createWebSocketChannel` and then call it using a declarative form `yield call(createWebSocketChannel)`?